packages/launchpad/src/components/Button.vue

```diff
@@ -1,45 +1,82 @@
 <template>
   <button
-    class="
-      border
-      px-4
-      py-2
-      rounded
-      text-sm
-      focus:border-indigo-600 focus:outline-transparent
-    "
-    :class="variantClasses + ' ' + additionalClasses"
-    @click="$emit('click')"
+    style="width: fit-content"
+    class="flex items-center border rounded-sm gap-8px focus:border-indigo-600 focus:outline-transparent"
+    :class="classes"
   >
-    <slot />
+    <span v-if="prefixIcon || $slots.prefix" :class="iconClasses" class="justify-self-start">
+      <slot name="prefix">
+        <Icon :icon="prefixIcon" :class="prefixIconClass" />
+      </slot>
+    </span>
+    <span class="flex-grow">
+      <slot />
+    </span>
+    <span v-if="suffixIcon || $slots.suffix" :class="iconClasses" class="justify-self-end">
+      <slot name="suffix">
+        <Icon :icon="suffixIcon" :class="suffixIconClass" />
+      </slot>
+    </span>
   </button>
 </template>
 
-<script lang="ts">
-import { defineComponent, PropType } from "vue";
+<script lang="ts" setup>
+import { defineProps, PropType } from "vue"
+import type { IconType } from '../types'
 
 const VariantClassesTable = {
   primary: "border-indigo-600 bg-indigo-600 text-white",
   outline: "border-gray-200 text-indigo-600",
   link: "border-transparent text-indigo-600",
-};
+}
 
-export default defineComponent({
-  emits: { click: null },
-  props: {
-    variant: {
-      type: String as PropType<"primary" | "outline" | "link">,
-      default: "primary",
-    },
-    class: {
-      type: String,
-      default: "",
-    },
-  },
-  setup(props) {
-    const variantClasses = VariantClassesTable[props.variant];
+const SizeClassesTable = {
+  sm: "px-1 py-1 text-xs",
+  md: 'px-2 py-1 text-xs',
+  lg: "px-4 py-2 text-sm",
+  xl: "px-6 py-3 text-lg"
+}
+
+const IconClassesTable = {
+  md: "h-1.25em w-1.25em",
+  lg: "h-2em w-2m",
+  xl: "h-2.5em w-2.5em"
+}
```

Comment:
nit: Could we make it so that these would be utility components? I see them being used in more than buttons.

Reply:
I'm not sure what the best way to abstract the design system sizes is. Some thought needs to be put into it.